app/src/main/java/at/bitfire/icsdroid/ProcessEventsTask.kt

```diff
@@ -77,7 +78,9 @@ class ProcessEventsTask(
             Log.d(Constants.TAG, "Removing all alarms from ${uid}: $this")
             alarms.clear()
         }
-        subscription.defaultAlarmMinutes?.let { minutes ->
+        val allDay = net.fortuna.ical4j.model.property.Duration(Duration.ofHours(24))
+        val isAllDay = event.duration == allDay
```

Comment:
All-day check: use `DateUtils.isDate(dtStart)`. Duration is not useful here, as it can be longer than 24h too; also days might not always have 24h here (unfortunately).